src/DynamoCore/Nodes/FunctionCallNodeController.cs

```diff
@@ -192,6 +193,38 @@ public virtual void SyncNodeWithDefinition(NodeModel model)
             InitializeOutputs(model);
             model.RegisterAllPorts();
             model.NickName = NickName;
+            OnSyncWithDefintionEnd(model);
+        }
+
+        /// <summary>
+        /// Event handler for the event when node starts syncing with its 
+        /// definition.
+        /// </summary>
+        public event Action<NodeModel> SyncWithDefinitionStart;
+        /// <summary>
+        /// Start syncing with its definition.
+        /// </summary>
+        /// <param name="obj"></param>
+        protected virtual void OnSyncWithDefinitionStart(NodeModel obj)
+        {
+            var handler = SyncWithDefinitionStart;
+            if (handler != null) handler(obj);
+        }
+
+
+        /// <summary>
+        /// Event handler for the event when node finishes syncing with its
+        /// definition.
+        /// </summary>
+        public event Action<NodeModel> SyncWithDefinitionEnd;
+        /// <summary>
+        /// Finish syncing with its definition.
+        /// </summary>
+        /// <param name="obj"></param>
+        protected virtual void OnSyncWithDefintionEnd(NodeModel obj)
+        {
+            var handler = SyncWithDefinitionEnd;
+            if (handler != null) handler(obj);
```

Comment:
For me personally I'd merge these into one (along with an `EventArgs` derived class) since they are nearly identical, but that's just me :)